src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -412,23 +416,61 @@ class DistributedClosestPointImpl
 
 public:
   /**
-   * Utility function to set the array of points
+   * Import object mesh points from the object blueprint mesh into internal memory.
    *
-   * \param [in] coords The root group of a mesh blueprint's coordinate values
+   * \param [in] mdMeshNode The blueprint mesh containing the object points.
+   * \param [in] valuesPath The path to the mesh points.
    * \note This function currently supports mesh blueprints with the "point" topology
    */
-  void importObjectPoints(const conduit::Node& coords, int nPts)
+  void importObjectPoints(const conduit::Node& mdMeshNode,
+                          const std::string& valuesPath)
   {
-    // Extract pointers to the coordinate data
-    // Note: The following assumes the coordinates are contiguous with stride NDIMS
-    // TODO: Generalize to support other strides
+    // TODO: See if some of the copies in this method can be optimized out.
 
-    // TODO: Add error checking for children 'x', 'y' and 'z' and striding
+    SLIC_ASSERT(sizeof(double) * DIM == sizeof(PointType));
 
-    // Copy the data into the point array of primal points
-    m_points.resize(nPts);
-    const std::size_t nbytes = sizeof(double) * DIM * nPts;
-    axom::copy(m_points.data(), coords["x"].data_ptr(), nbytes);
+    // Count points in the mesh.
+    int ptCount = 0;
+    for(const conduit::Node& domain : mdMeshNode.children())
+    {
+      auto& values = domain.fetch_existing(valuesPath);
+      const int N = internal::extractSize(values);
+      ptCount += N;
+    }
+
+    // Copy points to internal memory
+    PointArray coords(ptCount, ptCount);
+    axom::Array<axom::IndexType> domIds(ptCount, ptCount);
+    std::size_t copiedCount = 0;
+    conduit::Node tmpValues;
+    for(conduit::index_t d = 0; d < mdMeshNode.number_of_children(); ++d)
+    {
+      const conduit::Node& domain = mdMeshNode.child(d);
+
+      auto& values = domain.fetch_existing(valuesPath);
+
+      bool isInterleaved = conduit::blueprint::mcarray::is_interleaved(values);
+      if(!isInterleaved)
+      {
+        conduit::blueprint::mcarray::to_interleaved(values, tmpValues);
```

Comment:
Does `tmpValues` get cleared when calling `to_interleaved`?
If not, should it be declared within the `for` loop?

Reply:
It does get clear by `to_interleaved`, and I also explicitly clear it, just in case.